pkg/generators/snippets/mergemap.go

```diff
@@ -0,0 +1,35 @@
+package snippets
+
+import (
+	"errors"
+	"io"
+
+	"k8s.io/gengo/generator"
+)
+
+func GenerateMergeMapStringString(w io.Writer, ctx *generator.Context) error {
+	if ctx == nil {
+		return errors.New("nil pointer")
+	}
+
+	sw := generator.NewSnippetWriter(w, ctx, "$", "$")
+
+	sw.Do("// mergeMapStringString creates a new map and loads it from map args\n", nil)
+	sw.Do("// This function takes a least 2 args. Later map args take precedence.\n", nil)
+	sw.Do("func mergeMapStringString(m1 map[string]string, mapArgs ...map[string]string) map[string]string {\n", nil)
+	sw.Do("// populate initial map\n", nil)
+	sw.Do("outMap := map[string]string{}\n", nil)
+	sw.Do("for k, v := range m1 {\n", nil)
+	sw.Do("	outMap[k] = v\n", nil)
+	sw.Do("}\n", nil)
+	sw.Do("// iterate all args\n", nil)
+	sw.Do("for _, m := range mapArgs {\n", nil)
+	sw.Do("	for k, v := range m {\n", nil)
+	sw.Do("		outMap[k] = v\n", nil)
+	sw.Do("	}\n", nil)
```

Comment:
I added in tab spacing here (even though it shouldn't be necessary for final output) because I think it may help when reading this to see the loop structure. I'll try to be consistent in the other snippets code if we have nested for loops or logic.